.github/workflows/ci.yml

```
@@ -0,0 +1,57 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+      - '[0-9]+.[0-9]+'
+  pull_request:
+    branches:
+      - master
+      - '[0-9]+.[0-9]+'
+  schedule:
+    - cron:  '0 4 * * *'
+jobs:
+  Test:
+    name: Run tests
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        jdk: [8, 11]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: maven-java-${{ matrix.jdk }}
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install JDK
+        uses: joschi/setup-jdk@v2
+        with:
+          java-version: ${{ matrix.jdk }}
+      - name: Run tests
+        run: mvn -q clean verify -B
+  Deploy:
+    name: Deploy to OSSRH
+    if: ${{ github.repository_owner == 'vert-x3' && github.event_name == 'push' }}
+    needs: Test
+    runs-on: ubuntu-latest
+    env:
+      SONATYPE_NEXUS_USERNAME: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
+      SONATYPE_NEXUS_PASSWORD: ${{ secrets.SONATYPE_NEXUS_PASSWORD }}
+    steps:
+      - uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: maven-java-${{ matrix.jdk }}
```

```suggestion
          key: maven-java-8
```

`matrix.jdk` version does not exist in this job

---

why doesn't it exist ?

---

Because the matrix is defined in the `Test` job, not the `Deploy` job.

---

it seems to work though

---

It doesn't.

Take a look at this one for example: https://github.com/vert-x3/vertx-mail-client/actions/runs/393044498
The build passes yet the `Deploy` job is not executed (for no apparent reason).
It is due to the same problem.

---

I'll mass update the files we have asap

---

this is not executed because of the filter that only deploys on the push event, I think we should correct that to deploy also on a CRON event.

As you said it should use Java 8 for caching, however I pushed recently a commit and it deployed to OSSRH despite `${{ matrix.jdk }}` : https://github.com/vert-x3/vertx-unit/runs/1484890577?check_suite_focus=true .